script.metadata.editor/resources/lib/rating_updater.py

```diff
@@ -7,129 +7,181 @@
+        self.progressdialog.create('Updating', '')
+        self.total_items = total_items
+        self.processed_items = 0
+        self.progress = 0
 
-            UpdateRating({'dbid': item.get('%sid' % self.dbtype),
-                          'type': self.dbtype,
-                          'done_msg': False})
+    def canceled(self):
+        if RUN_IN_BACKGROUND:
+            return True if winprop('CancelRatingUpdater.bool') else False
+        else:
+            return True if self.progressdialog.iscanceled() or winprop('CancelRatingUpdater.bool') else False
 
-        progressdialog.close()
-        progressdialog = None
+    def update(self,cat,label):
+        self.processed_items += 1
+        progress = int(100 / self.total_items * self.processed_items)
+        processed = str(self.processed_items) + ' / ' + str(self.total_items)
 
-        notification(ADDON.getLocalizedString(32030), msg_text if msg_text else xbmc.getLocalizedString(19256))
+        if RUN_IN_BACKGROUND:
+            self.progressdialog.update(progress, processed, cat + ':' + label)
+        else:
+            self.progressdialog.update(progress, cat + ':[CR]' + label, processed)
 
-        winprop('UpdatingRatings', clear=True)
+    def close(self):
+        self.progressdialog.close()
+        self.progressdialog = None
+        winprop('CancelRatingUpdater', clear=True)
 
 
 class UpdateRating(object):
     def __init__(self,params):
         self.dbid = params.get('dbid')
         self.dbtype = params.get('type')
-        self.done_msg = True if params.get('done_msg', True) else False
         self.tmdb_type = 'movie' if self.dbtype == 'movie' else 'tv'
         self.tmdb_tv_status = None
         self.tmdb_mpaa = None
         self.tmdb_mpaa_fallback = None
+        self.tmdb_rating = None
+        self.imdb_rating = None
+        self.omdb_limit = False
         self.update_uniqueid = False
         self.episodeguide = None
 
-        self.method_details = 'VideoLibrary.Get%sDetails' % self.dbtype
-        self.method_setdetails = 'VideoLibrary.Set%sDetails' % self.dbtype
-        self.param = '%sid' % self.dbtype
-        self.key_details = '%sdetails' % self.dbtype
+        # collect db data
+        self.db = Database(dbid=self.dbid, dbtype=self.dbtype)
+        self.get_details()
 
-        self.init()
+        self.uniqueid = self.details.get('uniqueid', {})
+        self.ratings = self.details.get('ratings', {})
+        self.file = self.details.get('file')
+        self.year = self.details.get('year')
+        self.premiered = self.details.get('premiered') or self.details.get('firstaired')
+        self.title = self.details.get('title')
+        self.original_title = self.details.get('originaltitle') or self.title
+        self.tags = self.details.get('tag')
+
+        if ('thetvdb' or 'themoviedb') in self.details.get('episodeguide', ''):
```

Comment:
This is an incorrect construction. The expression in brackets always evaluates to `'thetvdb'`. If you need to check the presence of both strings, use `if 'thetvdb' in ... or 'themoviedb' in ... :` or a regular expression with `|` operator.

Reply:
Err. No idea what my brain did when I was writing this. Will be changed.